src/components/Tooltip/Tooltip.js

```diff
@@ -1,135 +1,169 @@
-import React from 'react';
 import PropTypes from 'prop-types';
 import styled, { css } from 'react-emotion';
+import { includes } from 'lodash';
 
-import { textKilo } from '../../styles/style-helpers';
+import {
+  textKilo,
+  centerAlign,
+  shadowSingle
+} from '../../styles/style-helpers';
 
 const CENTER = 'center';
+const TOP = 'top';
 const RIGHT = 'right';
+const BOTTOM = 'bottom';
 const LEFT = 'left';
 
-const baseStyles = css`
+const backgroundColor = ({ theme }) => theme.colors.n900;
+
+const baseStyles = ({ theme }) => css`
   label: tooltip;
-  position: relative;
   display: inline-block;
-
-  &:hover {
-    > span {
-      visibility: visible;
-      opacity: 1;
-    }
-  }
-`;
-
-const arrowAlignsStyles = ({ align, theme }) =>
-  ({
-    center: css`
-      label: tooltip__text-arrow--center;
-      &::after {
-        left: 50%;
-      }
-    `,
-    right: css`
-      label: tooltip__text-arrow--right;
-      &::after {
-        left: ${theme.spacings.mega};
-      }
-    `,
-    left: css`
-      label: tooltip__text-arrow--left;
-      &::after {
-        right: ${theme.spacings.mega};
-      }
-    `
-  }[align]);
-
-const tooltipTextAlignStyles = ({ align }) =>
-  ({
-    center: css`
-      label: tooltip__text--center;
-      left: 0;
-      right: 0;
-      margin-left: auto;
-      margin-right: auto;
-    `,
-    right: css`
-      label: tooltip__text--right;
-      left: 50%;
-    `,
-    left: css`
-      label: tooltip__text--left;
-      right: 50%;
-    `
-  }[align]);
-
-const tooltipTextStyles = ({ theme }) => css`
-  label: tooltip__text;
-  visibility: hidden;
-  opacity: 0;
-  width: max-content;
-  min-width: 60px;
-  background-color: ${theme.colors.black};
+  width: 100%;
+  max-width: 280px;
+  min-width: 120px;
+  background-color: ${backgroundColor({ theme })};
   color: ${theme.colors.white};
-  text-align: center;
   border-radius: ${theme.borderRadius.mega};
-  padding: ${theme.spacings.byte};
+  padding: ${theme.spacings.byte} ${theme.spacings.kilo};
   position: absolute;
   z-index: 1;
-  bottom: 125%;
   transition: opacity 0.3s;
   ${textKilo({ theme })};
+  ${shadowSingle({ theme })};
 
   &::after {
+    display: block;
     content: '';
+    width: 0;
+    height: 0;
     position: absolute;
-    margin-left: -${theme.spacings.bit};
-    border-width: ${theme.spacings.bit};
-    border-style: solid;
-    border-color: ${theme.colors.black} transparent transparent transparent;
-    top: 100%;
+    border: ${theme.spacings.byte} solid transparent;
   }
 `;
 
-const TooltipContainer = styled('span')`
-  ${tooltipTextStyles}
-  ${tooltipTextAlignStyles}
-  ${arrowAlignsStyles}
-`;
-const TooltipElement = styled('div')`
-  ${baseStyles};
+const oppositeMap = {
```

Comment:
Maybe keep calling it position map and add a comment to clarify -- if that's necessary at all.

Reply:
Eh. Then it would clash with the `position` prop. Naming variables is hard 🙈